src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -193,4 +198,16 @@ private CommandResult executeCommand(String commandText) throws CommandException
             throw e;
         }
     }
+
+    /**
+     * Refresh the person list panel and all person cards immediately
+     */
+    public static void refreshPersonListPanelImmediately() {
+        if (iNSTANCE == null) {
+            return;
+        }
+        iNSTANCE.personListPanelPlaceholder.getChildren().clear();
+        iNSTANCE.personListPanel = new PersonListPanel(iNSTANCE.logic.getFilteredPersonList());
+        iNSTANCE.personListPanelPlaceholder.getChildren().add(iNSTANCE.personListPanel.getRoot());
+    }
```

Comment:
Is this really necessary? If we can add and delete contact previously, then adding and deleting events should not require this extra method. Also, it may not be a huge issue, but it is also breaking abstraction by allowing command to directly interact with UI instead of using the model that was passed to it. Perhaps we can try using `EditCommand` to change the event list or `Model` to `setPerson`